crates/matrix-sdk-base/src/client.rs

```diff
@@ -122,9 +124,16 @@ impl BaseClient {
             #[cfg(feature = "e2e-encryption")]
             olm_machine: Default::default(),
             ignore_user_list_changes: Default::default(),
+            roominfo_update_sender: Arc::new(StdRwLock::new(None)),
         }
     }
 
+    /// Replaces the sender that triggers updates when room info changes. This
+    /// should only be used once at initialization.
+    pub fn set_roominfo_update_sender(&self, sender: broadcast::Sender<OwnedRoomId>) {
+        *self.roominfo_update_sender.write().unwrap() = Some(sender);
+    }
+
```

Comment:
Should not be needed. Actually, it _must_ be removed, because (i) it's a public method for an internal state, it's bad, (ii) if the sender changes, all receivers are broken, we don't want to change the sender.